### ISSUE TITLE
Traits 6.0.0rc0 compatibility

### ISSUE THREAD

**Issue by effigies** [nipy/nipype#3157]
### Summary

The `maint/1.3.x` branch is now [failing](https://travis-ci.org/nipy/nipype/builds/643996190) on the Python 3 `--pre` checks [with](https://travis-ci.org/nipy/nipype/jobs/643996204):

```
ImportError while loading conftest '/home/travis/build/nipy/nipype/nipype/conftest.py'.
nipype/__init__.py:56: in <module>
    from .pipeline import Node, MapNode, JoinNode, Workflow
nipype/pipeline/__init__.py:11: in <module>
    from .engine import Node, MapNode, JoinNode, Workflow
nipype/pipeline/engine/__init__.py:12: in <module>
    from .workflows import Workflow
nipype/pipeline/engine/workflows.py:27: in <module>
    from ...interfaces.base import (traits, TraitedSpec, TraitDictObject,
nipype/interfaces/__init__.py:13: in <module>
    from .io import DataGrabber, DataSink, SelectFiles, BIDSDataGrabber
nipype/interfaces/io.py:38: in <module>
    from .base import (
nipype/interfaces/base/__init__.py:14: in <module>
    from .core import (Interface, BaseInterface, SimpleInterface, CommandLine,
nipype/interfaces/base/core.py:41: in <module>
    from .traits_extension import traits, isdefined
nipype/interfaces/base/traits_extension.py:31: in <module>
    from traits.trait_handlers import TraitType, NoDefaultSpecified
E   ImportError: cannot import name 'TraitType'
```

Seems there's an API change. `TraitType` and `NoDefaultSpecified` are now defined in `traits.trait_type`, which did not previously exist. `TraitType` is part of `traits.api`, but `NoDefaultSpecified` is not.

Will submit fixes for 1.3.x and 1.4.x.

**Comment by effigies** [comment]
Might as well link to https://github.com/enthought/traits/pull/730, which produced this issue.

**Comment by effigies** [comment]
Fixed in #3158, #3159.
